rust/src/error.rs

```diff
@@ -284,17 +284,27 @@ mod tests {
     fn backtrace_returns_captured_backtrace() {
         let backtrace_string = b().unwrap_err().backtrace().to_string();
         let lines: Vec<_> = backtrace_string.lines().collect();
-        assert_eq!(
-            lines[0],
-            "   0: <carmen_rust::error::BTError<carmen_rust::error::Error> as core::convert::From<F>>::from",
-        );
-        assert!(lines[1].starts_with("             at ./src/error.rs"));
-        assert_eq!(
-            lines[2],
-            "   1: <core::result::Result<T,F> as core::ops::try_trait::FromResidual<core::result::Result<core::convert::Infallible,E>>>::from_residual"
-        );
-        assert_eq!(lines[4], "   2: carmen_rust::error::tests::b",);
-        assert!(lines[5].starts_with("             at ./src/error.rs"));
+        #[cfg(debug_assertions)]
```

Comment:
```suggestion
        // We use the presence of debug assertions as a proxy to detect whether this is a release build or not, since this affects the produced backtrace.
        // This will break if debug assertions are enabled for release builds.
        #[cfg(debug_assertions)]
```